compute/batch/backend.go

```diff
@@ -68,10 +79,52 @@ func (b *Backend) Submit(task *tes.Task) error {
 		req.ContainerOverrides.Vcpus = aws.Int64(vcpus)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	reqctx, cancel := context.WithTimeout(ctx, time.Second*60)
 	defer cancel()
 
-	_, err := b.client.SubmitJobWithContext(ctx, req)
+	resp, err := b.client.SubmitJobWithContext(reqctx, req)
+	if err != nil {
+		b.event.WriteEvent(ctx, events.NewState(task.Id, tes.SystemError))
+		b.event.WriteEvent(
+			ctx,
+			events.NewSystemLog(
+				task.Id, 0, 0, "error",
+				"error submitting task to AWSBatch",
+				map[string]string{"error": err.Error()},
+			),
+		)
+		return err
+	}
+
+	return b.event.WriteEvent(
+		ctx, events.NewMetadata(task.Id, 0, map[string]string{"awsbatch_id": *resp.JobId}),
+	)
+}
+
+// Cancel removes tasks from the AWS batch job queue.
+func (b *Backend) Cancel(ctx context.Context, taskID string) error {
+	var task *tes.Task
+	var err error
+
+	task, err = b.database.GetTask(
+		ctx, &tes.GetTaskRequest{Id: taskID, View: tes.TaskView_FULL},
+	)
+	if err != nil {
+		return err
+	}
+
+	// only cancel tasks in a QUEUED state
+	state := task.GetState()
+	if state != tes.State_QUEUED {
+		return nil
+	}
+
+	backendID := getAWSTaskID(task)
```

Comment:
There's a small chance that the batch ID isn't there. Probably want to handle that case.